feat: publish issue statistics

marnixdessing opened the pull request:
Calculate statistics nightly and monthly and store the results in a dynamoDB table. Results are then published on a public page part of the webapp.

- Fixes #203 
- Include frontend packages during install phase
- CalculateStatistics lambda that is triggered every night for daily calculations and the first of the month for monthly calculations
- Statistics page for retrieving and plotting the data

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/yivi-issue-app/actions/runs/9087991270

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #639 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/yivi-issue-app/actions/runs/9104442062

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #640 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/yivi-issue-app/actions/runs/9120947898

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #641 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/yivi-issue-app/actions/runs/9135768497

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #642 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

marnixdessing committed:
feat: basics for statistics page

marnixdessing committed:
feat: deploy dummy statistics page

marnixdessing committed:
chore: fixed test

marnixdessing committed:
Merge pull request #643 from GemeenteNijmegen/feat/publish-issue-events

feat: publish issue events front-end part

marnixdessing committed:
chore: use umd js for chartjs

marnixdessing committed:
chore: wrong file

marnixdessing committed:
chore: make sure scope changes work

marnixdessing committed:
chore: add statistics calculations

marnixdessing committed:
chore: updated resource counts in tests

marnixdessing committed:
chore: allow to start logs query

marnixdessing committed:
chore: pass correct url param to handler

marnixdessing committed:
chore: month timestamp does not have to include day

marnixdessing committed:
chore: add back calculations

marnixdessing committed:
chore: fetch actual data

marnixdessing committed:
chore: bugfixes

marnixdessing committed:
chore: promise after fetch

marnixdessing committed:
chore: set limit on items returned by query

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/yivi-issue-app/actions/runs/9151610680

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #645 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/yivi-issue-app/actions/runs/9166784513

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #646 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

marnixdessing committed:
chore: add logging

marnixdessing committed:
Merge branch 'acceptance' of github.com:GemeenteNijmegen/yivi-issue-app into acceptance

marnixdessing committed:
chore: add more logging to CalculateStatistics lambda

marnixdessing committed:
chore: it may be the case no results are returned

marnixdessing committed:
chore: if results is empty count is 0

marnixdessing committed:
Merge branch 'production' into acceptance

marnixdessing committed:
chore: remove datalabels form plot

marnixdessing committed:
chore: remove datalabels form plot by using object keys

marnixdessing committed:
chore: reverse query results oder

marnixdessing committed:
chore: plot data from lowest date to highest date

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/yivi-issue-app/actions/runs/9183307578

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #647 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

marnixdessing committed:
chore: calculate yearly statistics

marnixdessing committed:
chore: add annual data

marnixdessing committed:
chore: added explanation

marnixdessing committed:
chore: schedule annual statistics calculation

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/yivi-issue-app/actions/runs/9200190007

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #648 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

marnixdessing committed:
chore: log nog loggedin requests as well

marnixdessing committed:
Merge branch 'acceptance' of github.com:GemeenteNijmegen/yivi-issue-app into acceptance

joostvanderborg committed:
chore: Refactor statistics to construct

joostvanderborg commented on src/ApiStack.ts:
```diff
@@ -340,4 +370,75 @@ export class ApiStack extends Stack {
 
   }
 
+
```
De stack wordt nu erg lang, reactoren binnen de file wordt ook uitdagender zo. Is iets als dit een idee? https://github.com/GemeenteNijmegen/yivi-issue-app/compare/acceptance...chore/refactor-statistitcs?expand=1

joostvanderborg reviewed (dismissed):
Zie opmerking over refactoren stack, dat is nu nog mogelijk denk ik, als de boel is aangemaakt lastiger. Verder mooi om te delen!

marnixdessing committed:
Merge pull request #649 from GemeenteNijmegen/chore/refactor-statistitcs

chore: Refactor statistics to construct

joostvanderborg committed:
fix: Pass table back to stack

marnixdessing committed:
Merge pull request #650 from GemeenteNijmegen/chore/refactor-statistitcs

fix: Pass table back to stack

joostvanderborg committed:
feat: Add statistics as table

The canvas-based chart is inaccessible to screen
readers etc.. This adds the same data as a table,
to provide a text-based alternative.

joostvanderborg committed:
fix: Return fetch to working order

joostvanderborg reviewed (dismissed):
Zie eerdere approval

marnixdessing committed:
Merge pull request #652 from GemeenteNijmegen/feat/stats-as-table

feat: stats as table

marnixdessing committed:
chore: missing } after function body
